src/Altinn.Platform/Altinn.Platform.Events/Events/Controllers/PushController.cs

```diff
@@ -0,0 +1,142 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.Platform.Events.Controllers
+{
+    /// <summary>
+    /// Controller responsible for pushing eventsto subscribers
+    /// </summary>
+    [Route("events/api/v1/push")]
+    [ApiController]
+    public class PushController : ControllerBase
+    {
+        private readonly ISubscriptionService _subscriptionService;
+        private readonly IRegisterService _registerService;
+        private readonly IEventsService _eventsService;
+
+        private readonly ILogger _logger;
+        private readonly string _eventsBaseUri;
+        private readonly AuthorizationHelper _authorizationHelper;
+        private readonly AccessTokenSettings _accessTokenSettings;
+
+        private const string DefaultIssuer = "Altinn";
+        private const string DefaultType = "string";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PushController"/> class.
+        /// </summary>
+        public PushController(
+        IEventsService eventsService,
+        ISubscriptionService subscriptionService,
+        IRegisterService registerService,
+        IOptions<GeneralSettings> settings,
+        ILogger<EventsController> logger,
+        IPDP pdp,
+        IOptions<AccessTokenSettings> accessTokenSettings)
+        {
+            _eventsService = eventsService;
+            _registerService = registerService;
+            _logger = logger;
+            _subscriptionService = subscriptionService;
+            _eventsBaseUri = $"https://platform.{settings.Value.Hostname}";
+            _authorizationHelper = new AuthorizationHelper(pdp);
+            _accessTokenSettings = accessTokenSettings.Value;
+        }
+
+        /// <summary>
+        /// Inserts a new event.
+        /// </summary>
+        /// <returns>The application metadata object.</returns>
+        [Authorize]
+        [HttpPost]
+        [Consumes("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Produces("application/json")]
+        public async Task<ActionResult<string>> Post([FromBody] CloudEvent cloudEvent)
+        {
+            string sourceFilter = GetSourceFilter(cloudEvent.Source);
+
+            if (!string.IsNullOrEmpty(sourceFilter))
+            {
+                List<Subscription> orgSubscriptions = await MatchOrgSubscriptions(sourceFilter, cloudEvent.Subject, cloudEvent.Type);
+                foreach (Subscription subscription in orgSubscriptions)
+                {
+                    await AuthorizeAndPush(cloudEvent, subscription);
+                }
+
+                List<Subscription> subscriptions = await MatchSubscriptionExcludeOrgs(sourceFilter, cloudEvent.Subject, cloudEvent.Type);
+                foreach (Subscription subscription in subscriptions)
+                {
+                    await AuthorizeAndPush(cloudEvent, subscription);
```

Comment:
Extracted code in private method

Reply:
I still don't quite understand why the Org subscriptions and Other subscriptions are separated here. Both lists are handled by the same method.

Reply:
But the list is different. Two different calls to repository with different input.

Reply:
I understand that org subscription can be handled faster by authorization than the other subscriptions, but the code we have here do not handle them differently. It could have been a single list. Any separation of logic happens further down on individual subscriptions. At least that is what it looks like to me.

Reply:
@SandGrainOne now I got your point. The explanation is that we will add caching to the org list. 
I belive it is more readable now. But I understand it could be different views on that.